apis/kueue/v1beta1/workload_types.go

```diff
@@ -59,6 +59,15 @@ type WorkloadSpec struct {
 	// +kubebuilder:default=""
 	// +kubebuilder:validation:Enum=kueue.x-k8s.io/workloadpriorityclass;scheduling.k8s.io/priorityclass;""
 	PriorityClassSource string `json:"priorityClassSource,omitempty"`
+
+	// QueueingPolicy that will determine if a job needs to be requeued or not after being suspended
```

Comment:
```suggestion
	// ReQueueingPolicy that will determine if a job needs to be requeued or not after being suspended
```

Also, `QueuingPolicy` sounds like the queuing method can be changed regardless of whether the job is suspended or not.

So, we should mention as an API name that this policy will apply to the job after jobs are admitted at least once.

WDYT? @alculquicondor @kerthcet

Reply:
> So, we should mention as an API name that this policy will apply to the job after jobs are admitted at least once.

I think in most cases the use-case will be suspended already-admitted jobs, but you can have a job that is not yet admitted (due to no avaialble resources) and then have queueing disabled

Reply:
> Also, QueuingPolicy sounds like the queuing method can be changed regardless of whether the job is suspended or not.

Yes, this is true as of the current implementation. If you disable for suspended job, it will be suspended forever. If you disable for a running job, it will NOT suspend the job but disable re-queueing if manually suspended

Reply:
> but you can have a job that is not yet admitted (due to no available resources) and then have queueing disabled

Currently, we can disable queuing by removing the queue name label from the job. 
So I think that it will cause confusion for users to support the same feature in multiple ways.

Reply:
I found the `QueueingPolicy=Never` might be useful on the workflow use case: https://github.com/kubernetes-sigs/kueue/issues/1091#issuecomment-1789324597.

However, if we introduce job readiness gates (https://github.com/kubernetes/kubernetes/issues/121681), we have similar features on both the kueue side and the job-controller side.

Reply:
NVM

Maybe `QueueingPolicy=Never` and Job readiness gates can co-exist.

Reply:
I think so, we still need a representation in the Workload API.

Reply:
Yes, I imagined the following steps:

1. The external controller adds a gate to the job.
2. jobframework reconciler creates workload with `QueueingPolicy=Never`.
3. The external controller deletes a gate to the job.
4. jobframework reconciler updates workload with `QueueingPolicy=ALways`.